abci/types/mocks/base.go

```diff
@@ -0,0 +1,182 @@
+package mocks
+
+import (
+	types "github.com/tendermint/tendermint/abci/types"
+)
+
+type BaseMock struct {
+	base *types.BaseApplication
+	*Application
```

Comment:
Need this be exported? It looks like we've got all the methods explicitly implemented anyway. Does the caller need to retrieve the wrapped application?

(If so: I'd suggest we make them pass it to the constructor. If not: we can probably unexport, unless I missed something)

Reply:
This `Application` contains the mocking methods, eg. `.On("function name"...)`, so it needs to be public to allow users of this type to properly mock.

Reply:
Ah, I see, you're having them add the methods _after_ constructing the base. I was thinking they'd set up the app and pass it as an argument. But that works too.